packages/query-core/src/notifyManager.ts

```diff
@@ -19,6 +21,23 @@ export function createNotifyManager() {
   let batchNotifyFn: BatchNotifyFunction = (callback: () => void) => {
     callback()
   }
+  let scheduleFn: ScheduleFunction
+
+  const setBatchMethod = (method: BatchMethod) => {
+    if (method.type === "tick") {
+      scheduleFn = queueMicrotask
+    } else if (method.type === "raf") {
+      scheduleFn = requestAnimationFrame
+    } else if (method.type === "timer") {
+      scheduleFn = (cb) => setTimeout(cb, method.timeout ?? 0)
+      // Disable because condition is necessary for typescript to detect method.fn
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    } else if (method.type === "custom") {
+      scheduleFn = method.fn
+    }
+  }
```

Comment:
on a second thought - while the api of rtk-query is nice, wouldn't we able to achieve the exact same thing only with the custom type?

- timeout: `setBatchMethod({ type: 'custom', fn: (cb) => setTimeout(cb, 0) })`
- raf: `setBatchMethod({ type: 'custom', fn: requestAnimationFrame })`
- tick: `setBatchMethod({ type: 'custom', fn: queueMicrotask })` 

with that in mind, we could simplify our api and just offer:

```
setBatchMethod: (callback: () => void) => void
```

and show examples in the docs on how to implement it with timeout / raf / tick ?

@markerikson maybe you can help us shed some light on how you decided on that specific api for the autobatchenhancer ?

Reply:
Not sure there was much thinking put into it. My initial implementation was hardcoded to `queueMicrotask`, and then I realized it might be good to make it configurable. I wanted to build in the implementations so users didn't have to write the callback implementations, so I opted for the enum / literal option approach.

Reply:
thanks. in that case, @NathanHuisman could you maybe refactor towards this syntax?

```
setBatchMethod: (callback: () => void) => void
```

because for raf / tick, it would just be:

```
notifyManager.setBatchMethod(requestAnimationFrame)
notifyManager.setBatchMethod(queueMicrotask)
```

the only syntax that would be a bit more to write would be `setTimeout` with a different value, but that's fine by me given that this gives us a smaller api surface and less code on our end.